crates/router/src/configs/settings.rs

```diff
@@ -242,6 +243,15 @@ pub struct DummyConnector {
     pub discord_invite_url: String,
 }
 
+#[derive(Debug, Deserialize, Clone)]
+pub struct CorsSettings {
+    #[serde(deserialize_with = "deserialize_hashset")]
+    pub origins: HashSet<String>,
```

Comment:
Would this allow us to specify `*`? What if someone self-hosting wants to allow all origins?